### PR TITLE
feat: enable FIPS compliance by switching to ubi9 minimal and installing openssl; add Infrastructure Feature Test annotations (RHIDP-294)

### PR DESCRIPTION
**Pull request by nickboldt**
### What does this PR do?

feat: enable FIPS compliance by switching to ubi9 minimal and installing openssl; add Infrastructure Feature Test annotations ([RHIDP-294](https://issues.redhat.com//browse/RHIDP-294))

Signed-off-by: RHDH Build (rhdh-bot) <rhdh-bot@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
N/A (or see commit message above for issue number)

### How to test this PR?
N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works, if one exists](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections What issues does this PR fix or reference and How to test this PR completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by nickboldt** [comment]
FIPS compliance checked with build https://brewweb.engineering.redhat.com/brew/taskinfo?taskID=58051064

```
#➔ ~/4/check-payload/check-payload scan image --spec  registry-proxy.engineering.redhat.com/rh-osbs/rhdh-rhdh-rhel9-operator:rhdh-1-rhel-9-containers-candidate-69291-20240112183533
I0112 15:13:12.322240   43446 main.go:100] "scan" version="0.3.0-4-g362bde27"
---- Warning Report
+-------------------------+-----------------+-----------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------+
| OPERATOR NAME           | EXECUTABLE NAME | STATUS                                                          | IMAGE                                                                                                                          |
+-------------------------+-----------------+-----------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------+
| rhdh-operator-container | /manager        | go binary has no build tags set (should have strictfipsruntime) | registry-proxy.engineering.redhat.com/rh-osbs/rhdh-rhdh-rhel9-operator:rhdh-1-rhel-9-containers-candidate-69291-20240112183533 |
+-------------------------+-----------------+-----------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------+
---- Successful run with warnings
```

File: docker/Dockerfile [hunk 1]
```diff
@@ -52,9 +52,11 @@ RUN go mod download
 RUN export ARCH="$(uname -m)" && if [[ ${ARCH} == "x86_64" ]]; then export ARCH="amd64"; elif [[ ${ARCH} == "aarch64" ]]; then export ARCH="arm64"; fi && \
     CGO_ENABLED=1 GOOS=linux GOARCH=${ARCH} go build -a -o manager main.go
 
-# NOTE: ubi-micro will not be FIPS compliant, if openssl is not installed
-#@follow_tag(registry.redhat.io/ubi9/ubi-micro:latest)
-FROM registry.access.redhat.com/ubi9/ubi-micro:9.3-9
+# Install openssl for FIPS support
```
Comment (gazarenkov) [inline docker/Dockerfile:55]: Curious, if we need it upstream, looks like it about 5 times bigger than ubi-micro...

**Comment by nickboldt** [comment]
By my math it's only 2x bigger https://quay.io/repository/janus-idp/operator?tab=tags ... adding only 32M to the compressed container image.

**Comment by gazarenkov** [comment]
Yes, it somehow fits the data that image size grown from 6.9M to 36M (operator is like 22-25M).
Curious, if we need support FIPS upstream.

**Comment by nickboldt** [comment]
Counter argument: Why do we want the upstream operator code to be different from the downstream one? Do we want 2x the testing and support burden?

**Pull request merged**